dist/stat_data.py

```diff
@@ -335,6 +339,41 @@ def __init__(self, name, desc, flags=''):
     CapacityStat('fsync_all_fh_total', 'background fsync file handles considered'),
     CapacityStat('fsync_all_time', 'background fsync time (msecs)', 'no_clear,no_scale'),
 
+    ##########################################
+    # Checkpoint statistics
+    ##########################################
+    CheckpointStat('checkpoints', 'transaction checkpoints'),
+    CheckpointStat('checkpoint_custom_dsrc', 'number of custom data sources checkpointed'),
+    CheckpointStat('checkpoint_fsync_post', 'transaction fsync calls for checkpoint after allocating the transaction ID'),
+    CheckpointStat('checkpoint_fsync_post_duration', 'transaction fsync duration for checkpoint after allocating the transaction ID (usecs)', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_generation', 'transaction checkpoint generation', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_handle_applied', 'transaction checkpoint most recent handles applied'),
+    CheckpointStat('checkpoint_handle_checked', 'transaction checkpoint data handle operation applied'),
+    CheckpointStat('checkpoint_handle_duration', 'transaction checkpoint most recent duration for gathering all handles (usecs)', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_handle_duration_apply', 'transaction checkpoint most recent duration for gathering applied handles (usecs)', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_handle_duration_skip', 'transaction checkpoint most recent duration for gathering skipped handles (usecs)', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_handle_skipped', 'transaction checkpoint most recent handles skipped'),
+    CheckpointStat('checkpoint_handle_walked', 'transaction checkpoint most recent handles walked'),
+    CheckpointStat('checkpoint_prep_max', 'transaction checkpoint prepare max time (msecs)', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_prep_min', 'transaction checkpoint prepare min time (msecs)', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_prep_recent', 'transaction checkpoint prepare most recent time (msecs)', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_prep_running', 'transaction checkpoint prepare currently running', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_prep_total', 'transaction checkpoint prepare total time (msecs)', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_presync', 'number of handles visited after checkpoint writes complete'),
+    CheckpointStat('checkpoint_scrub_target', 'transaction checkpoint scrub dirty target', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_scrub_time', 'transaction checkpoint scrub time (msecs)', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_skipped', 'transaction checkpoints skipped because database was clean'),
+    CheckpointStat('checkpoint_state', 'last potentially expensive operation attempted', 'no_clear,no_scale'),
```

Comment:
It isn't clear what the content of this statistic would be (before reading the code). I know that TSEs know to look for the old `checkpoint running` stat. It needs to be really clear what replaces that.

Reply:
Right, would it be better to say something like "non-zero means checkpoint is running"? I could also leave the old "checkpoint running" in, but it feels redundant with the state.

Reply:
I like the idea, I think the description could be improved. The TSEs don't see the statistic variable name, only the description, so something like "transaction checkpoint progress state" (see comment above about keeping the category string in the description)

Reply:
You're right, the description is poor right now. I'll fix it to "progress state", modulo my comment in the other thread about description prefixes.

Reply:
Done in ae54606